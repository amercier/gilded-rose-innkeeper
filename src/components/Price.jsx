import React from 'react';
import { number } from 'prop-types';
import { Tooltip } from 'antd';
import styled from 'styled-components';
import humanFormat from 'human-format';
import formatNumber from 'simple-format-number';

const Info = styled.span`
  border-bottom: 1px dotted;
  cursor: help;
`;

/**
 * Display a price and an indicator about its trend.
 *
 * @param {Object} props - React component properties.
 * @param {number} props.value - Price value.
 * @returns {React.Element} The rendered element.
 */
const Price = ({ value }) => {
  if (value < 1000) {
    return value;
  }

  return (
    <Tooltip
      title={formatNumber(value, {
        fractionDigits: 0,
        symbols: { grouping: ' ' },
      })}
    >
      <Info>{humanFormat(value, { decimals: 1, separator: '' })}</Info>
    </Tooltip>
  );
};

Price.propTypes = {
  value: number.isRequired,
};

export default Price;
