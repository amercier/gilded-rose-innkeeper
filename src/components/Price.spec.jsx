import React from 'react';
import { shallow } from 'enzyme';
import Price from './Price';

describe('Price', () => {
  it('renders the value for small numbers', () => {
    [0, 10, 100, 999].forEach(smallNumber => {
      const price = shallow(<Price value={smallNumber} />);
      expect(price.text()).toBe(`${smallNumber}`);
    });
  });

  it('renders a tooltip for large numbers', () => {
    [1000, 123456, 123456789].forEach(largeNumber => {
      const price = shallow(<Price value={largeNumber} />);
      expect(price).toMatchSnapshot();
    });
  });
});
