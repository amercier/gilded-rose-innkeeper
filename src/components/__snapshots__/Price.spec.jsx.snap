// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price renders a tooltip for large numbers 1`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  placement="top"
  title="1 000"
  transitionName="zoom-big-fast"
>
  <styled.span>
    1k
  </styled.span>
</Tooltip>
`;

exports[`Price renders a tooltip for large numbers 2`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  placement="top"
  title="123 456"
  transitionName="zoom-big-fast"
>
  <styled.span>
    123.5k
  </styled.span>
</Tooltip>
`;

exports[`Price renders a tooltip for large numbers 3`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  placement="top"
  title="123 456 789"
  transitionName="zoom-big-fast"
>
  <styled.span>
    123.5M
  </styled.span>
</Tooltip>
`;
